design/backup-resources-order.md

```diff
@@ -0,0 +1,39 @@
+## Backup Resources Order
+This document proposes a solution that allows user to specify a backup order for resources of specific resource type.
+
+## Background
+During backup process, user may need to back up resources of specific type in some specific order to ensure the resources were backup properly because these resources are related and ordering might be required to preserve the consistency for the apps to recover itself �from the backup image 
+(Ex: master-slave database pods in a cluster).
+
+## Goals
+- Enable user to specify an order of back up resources belong to specific resource type
+
+## Alternatives Considered
+- Use a plugin to backup an resources and all the sub resources.  For example use a plugin for StatefulSet and backup pods belong to the StatefulSet in specific order.  This plugin solution is not generic and requires plugin for each resource type.
+
+## High-Level Design
+User will specify a map of resource type to list resource names (separate by semicolons).  Each name will be in the format "namespaceName/resourceName" to enable ordering accross namespaces.  Based on this map, the resources of each resource type will be sorted by the order specified in the list of resources.  If a resource instance belong to that specific type but its name is not in the order list, then it will be put behind other resources that are in the list.

```

Comment:
If `resourceName` is specified, but does not exist in the cluster, what should the behavior be? Generally, Velero has been treating these as cluster problems and issues a warning, but completes the backup operation successfully. I'm curious if that's acceptable from your use case.

Reply:
In my usage, we will first fetch the list of pods from k8s API Server using app label selector, then processes this list to generate a sequence of pods to backup to ensure App Consistent backup so the pods should exist.  However, in general, I agree that we should give a warning as you mentioned if the resource doesn't exist.